### ISSUE TITLE
TCP connections from restify server spikes occasionally for unknown reasons

### ISSUE THREAD

**Issue by serv** [restify/node-restify#1022]
TLDR
The number of TCP connections from our restify servers to downstream services spike occasionally and I am not sure what could be causing this. See the graph below. Note the spike in the number of TCP connections going up to 300k, when the normal number is around 1k. 

Any idea what might be causing this?

<img width="998" alt="screen shot 2016-03-04 at 3 56 36 pm" src="https://cloud.githubusercontent.com/assets/103456/13543647/ed6cb1e0-e221-11e5-8ea9-6e381259e0a5.png">

Here's what I found so far.

1. The whole series of event starts with an error message like this. I think this is some temporary hiccup in our network (although I am not sure why this is happening). 
`"err":{"message":"read ECONNRESET","stack":"Error: read ECONNRESET\nat errnoException (net.js:901:11)\nat TCP.onread (net.js:556:19)","name":"Error"}`

2. TCP connections to the downstream starts to bubble up very rapidly.

We use `createStringClient()` and `createJsonClient()` to create clients to talk to the downstream services. 

Our restify version is 2.8.5. Our node version is 0.10.38. We are planning to upgrade these two soon.

**Comment by micahr** [comment]
@serv - This seems like something within your infrastructure or application. The error message you posted comes from Node directly, and isn't Restify specific. TCP connections could also start stacking up for a variety of reasons. If you find something specific within Restify that causes this issue, please let us know; but we aren't able to provide enterprise level support for Restify and it's use within specific environments at this time.

**Comment by micahr** [comment]
Was there a resolution within your environment / application?

**Comment by serv** [comment]
Sorry for the late reply. It was our application's fault. We received a call that actually attempts to make over 20k calls to our downstream services. Because there are so many sockets available to use (around 16k), the call fails eventually. But that machine goes down because the memory runs out with 16k outstanding TCP connections. Client attempts to make the same call again. Brings down the next machine. So on and so forth.
